### PR TITLE
Update C++, Arduino, and MATLAB API Docs

### PR DESCRIPTION
**Pull request by vertiq-jordan**
Pretty much what the title says. I've updated the C++, Arduino, and MATLAB documentation to match what we did with the new Python docs. Now, there's some more examples and explanations of why things are how they are, install instructions, etc.

I also updated the copyright year to 2024 while I was in there.

Built docs are here: https://iqmotion.readthedocs.io/en/feature-add_cpp_arduino_matlab/

### CODE REVIEW

File: docs/api_docs/matlab.rst [hunk 1]
```diff
@@ -1,66 +1,153 @@
...
+Each MATLAB client is linked with a json file stored in a `git submodule <https://github.com/iq-motion-control/client_files/tree/74284e59abbe86e2b6d629a3727163a28254fd70>`_ that specifies the entries within each client.
 
-``data_all = client_object.get_all()``
+In order to create an instance of a client object, you must have at least a ``MessageInterface`` (``com``), and optionally a target Module ID (``obj_id``) 
+and timeout (``timeout``). By default the MATLAB API assumes a Module ID of 0 and a timeout of 0.2.
 
-performs a get on all messages in list and stores it in data all.
+Suppose we wanted to create an instance of the Microcontroller Temperature Monitor that uses our ``com`` instance created above, 
+communicates with a module with Module ID 42, and we allow up to 0.5 seconds for responses. To do so, we'd create an instance 
+of ``TemperatureMonitorUcClient`` with ``uc_temp = TemperatureMonitorUcClient(‘com’, com, ‘obj_id’, 42, ‘timeout’, 0.5);``
```
Comment (iq-fred) [inline docs/api_docs/matlab.rst:59]: Also maybe good to break out to a little code block, I find people often just skim through for code blocks and pictures, so having important stuff pop out in blocks can be helpful

File: docs/api_docs/matlab.rst [hunk 2]
```diff
@@ -1,66 +1,153 @@
...
+
+.. image:: ../_static/api_pics/get_timeout.png
+
+Save
+^^^^^^
+All MATLAB save commands have the format ``client_object.save("client_entry")``.
+
+The save function takes the currently set entry value, and stores it in the module's persistent memory. Values that are saved are retained on power cycles.
+
+Suppose we want to save the timeout value set above. To do so, run ``prop.save('timeout')``.
```
Comment (iq-fred) [inline docs/api_docs/matlab.rst:119]: I know its a bit silly, but popping this out to a code block or picture may be good for consistency/eye-catching

File: docs/api_docs/matlab.rst [hunk 3]
```diff
@@ -1,66 +1,153 @@
...
+    %Check our temperature
+    uc_temp.get("uc_temp")
+
+    %Loop forever
+    while 1
+        %Spin slowly
+        prop_control.set("ctrl_velocity", 20);
+
+        %Check our velocity now
+        brushless_drive.get("obs_velocity")
```
Comment (iq-fred) [inline docs/api_docs/matlab.rst:150]: Same thing as above about printing

File: docs/api_docs/matlab.rst [hunk 4]
```diff
@@ -1,66 +1,153 @@
...
+    %Create a MessageInterface to handle IQUART communication
+    com = MessageInterface("COM3", 115200);
+
+    %Create the clients that we need (all using default Module ID of 0)
+    uc_temp = TemperatureMonitorUcClient('com', com);
+    prop_control = PropellerMotorControlClient('com', com);
+    brushless_drive = BrushlessDriveClient('com', com);
+
+    %Check our temperature
+    uc_temp.get("uc_temp")
```
Comment (iq-fred) [inline docs/api_docs/matlab.rst:142]: Should we print this out, or does Matlab already do that when you get?

**Review by iq-fred — CHANGES_REQUESTED** [review]

File: docs/api_docs/matlab.rst (hunk 4 above)
Comment (vertiq-jordan) [reply docs/api_docs/matlab.rst:142]: if you don't add a semi colon in matlab, it prints the result.

Comment (iq-fred) [reply docs/api_docs/matlab.rst:154]: Magical

**Review by iq-fred — CHANGES_REQUESTED** [review]

**Review by iq-fred — APPROVED** [review]

**Pull request merged**